### ISSUE TITLE
No edge dragging commences when trying

### ISSUE THREAD

**Issue by buffpojken** [cytoscape/cytoscape.js-edgehandles#42]
I think 2.6.0 broke the extension - completely unchanged code stopped working between 2.5.0 and 2.6.0. 

Initialisation works just fine, but when trying to actually drag out an edge to connect you start to drag the node instead. Regressing to 2.5.0 restores expected behaviour. 

I'll look into this on monday unless someone has a solid idea on why already, but I'm guessing the jQuery-like shim now used instead has some slight incompatibility with how events used to be handled (based on skimming the commits in 30 seconds, treat accordingly. )

**Comment by maxkfranz** [comment]
Would you please paste your options?  I tried different ones with 2.6 before release, but it could be a particular combination is problematic...

**Comment by buffpojken** [comment]
Sure thing, here goes: 

```
			e.cy.edgehandles({
				complete: this.options.onEdgeAdded, 
				toggleOffOnLeave: true,
				handleNodes: "node[^group]",
				handleSize: 34,
				handleIcon: edgehandle,
				handleColor: '#DF0085',
				edgeType: (source, target) => {
					let cycle = e.cy.elements().aStar({root: target, goal: source, directed: true}); 
					return (cycle.found ? false : 'flat');
				}, 
				edgeParams: this.options.edgeParams.bind(this)
			});	
```

**Comment by maxkfranz** [comment]
I tried these options (similar to yours w/o app refs) and it worked for me on demo.html:

```js
cy.edgehandles({
  toggleOffOnLeave: true,
  handleNodes: "node",
  handleSize: 34,
  handleColor: '#DF0085',
  handleIcon: (function(){
    var img = new Image();

    img.src = '../cytoscape.js/debug/images/gnu.svg';
    img.width = 34;
    img.height = 34;

    return img;
  })(),
  edgeType: function(){ return 'flat'; }
});
```

Do these options work for you?  Are you able to isolate the problematic option?  

Thanks

**Comment by buffpojken** [comment]
No, these do not work. 

These exact options provides the node-drag behavior instead: 

````
 e.cy.edgehandles({
   toggleOffOnLeave: true,
   handleNodes: "node",
   handleSize: 34,
   handleColor: '#DF0085',
   handleIcon: false,
   edgeType: function(){ return 'flat'; }
 });
```

**Comment by emelendez** [comment]
Same issue here.
Once you click/drag over rounded handle, this rounded handle disappear, hence you drag the node but no edge line appears on screen.

It seems as if this rounded handle looses focus.

**Comment by maxkfranz** [comment]
The only changes to 2.6 are @tongbin's [pull request](https://github.com/cytoscape/cytoscape.js-edgehandles/pull/41).

@buffpojken's options work for me on the debug page.  Maybe you're using other calls as well?

I looked it over, and the only thing I noticed was that his jquery shim is missing event namespace support.  I'm not sure if that's the issue, but probably @tongbin would know more.

**Comment by buffpojken** [comment]
Hmm, I just had an idea. I'm installing edgehandles via bower - and the resolution listed in .bower.json doesn't match the release tagged 2.6.0 here on Github? 

Is there perhaps a conflict hiding here somewhere?

**Comment by buffpojken** [comment]
More investigation proves it's the jQuery shim breaking something. Commenting it out and using regular jQuery works just fine. I'll continue to investigate.

**Comment by buffpojken** [comment]
Okey, so further investigation proves that the difference between running with the shim and without is that the if-clause on #973:

``` if( Math.abs( x - hx ) > hrTarget || Math.abs( y - hy ) > hrTarget ) {```

triggers even if on the handle with the shim, but not without it - thus cancelling the action before doing anything.

**Comment by buffpojken** [comment]
I have found the issue - the jQuery shim has a major bug when calculating offset - it only takes the immediate parent into account, not the true offset from the window. 

My canvas is wrapped in a div, which is turn placed 50px below the window top, and all coordinates used for determining if the mouse is actually on top of the handle are off by 50px along the Y-axis. I have also confirmed this by manually comparing the offset reported by the shim versus the offset reported by jQuery. 

PR incoming.

**Comment by buffpojken** [comment]
@maxkfranz - this also explains why it worked fine on your demo page - because I guess the canvas isn't offset from top-left in that? The old offset management would work perfectly fine then :)

**Comment by tongbin** [comment]
Thanks a lot for your guys' testings, I'll handle this issue right now.

**Comment by maxkfranz** [comment]
Merged PR and added improvements to #41

Please confirm that my changes in #41 don't break anything in your cases.  My changes should help make it easier to reuse the shim in other extensions
